services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/extensions/VersionedRecordExtensionTest.java

```diff
@@ -168,15 +178,344 @@ public void beforeWrite_returnsNoOpModification_ifVersionAttributeNotDefined() {
     @Test(expected = IllegalArgumentException.class)
     public void beforeWrite_throwsIllegalArgumentException_ifVersionAttributeIsWrongType() {
         FakeItem fakeItem = createUniqueFakeItem();
-        Map<String, AttributeValue> fakeItemWIthBadVersion =
+        Map<String, AttributeValue> fakeItemWithBadVersion =
             new HashMap<>(FakeItem.getTableSchema().itemToMap(fakeItem, true));
-        fakeItemWIthBadVersion.put("version", AttributeValue.builder().s("14").build());
+        fakeItemWithBadVersion.put("version", AttributeValue.builder().s("14").build());
 
         versionedRecordExtension.beforeWrite(
             DefaultDynamoDbExtensionContext.builder()
-                                           .items(fakeItemWIthBadVersion)
+                                           .items(fakeItemWithBadVersion)
                                            .operationContext(PRIMARY_CONTEXT)
                                            .tableMetadata(FakeItem.getTableMetadata())
                                            .build());
     }
+
+    @Test
+    public void beforeWrite_versionEqualsStartAt_treatedAsInitialVersion() {
```

Comment:
Can we add tests for different table schemas such as static table schema, immutable table scheme and bean table schema? https://github.com/aws/aws-sdk-java-v2/pull/6067/files#diff-9826429ca566bd1325f9ffa5d4fe1141cd26df01ec67262bab6f4acb12c1aa28